### PR TITLE
Add scheduler job executor script

### PR DESCRIPTION
**Pull request by vamahaja**
Add scheduled job executor groovy script. It perform bellow tasks -

1. Check for recipe file modification time.
2. Get RHCEPH content and RHCEPH version.
3. Execute jobs based on modified recipe files.

### CODE REVIEW

**Review by manasagowri — COMMENTED** [review]
Looks good to me. Can you please update the logs of one execution when you test this.

**Review by manasagowri — CHANGES_REQUESTED** [review]

File: pipeline/scheduled-pipeline-executor.groovy [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (!validRecipeFiles) {
+            echo "There no new RHCS build since last Friday"
+            exit 1
+        }
+
+        for (validRecipeFile in validRecipeFiles) {
+            def recipeContent = sh(
+                    returnStdout: true,
+                    script: "${SSHServer} \"yq e '.latest' ${validRecipeFile}\"").trim()
+            def rhcephVersion = sh(
```
[the remark this answers is not in the document]
Comment (vamahaja) [reply pipeline/scheduled-pipeline-executor.groovy]: Done

File: pipeline/scheduled-pipeline-executor.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        if (!validRecipeFiles) {
+            throw new RuntimeException("There no new RHCS build since last Friday")
+        }
+
+        for (validRecipeFile in validRecipeFiles) {
+            def rhcephVersion = validRecipeFile.split("/").last().replace(".yaml", "")
+            def recipeContent = sh(
+                    returnStdout: true,
+                    script: "${SSHServer} \"yq e '.latest' ${validRecipeFile}\"").trim()
```
Comment (psathyan) [inline pipeline/scheduled-pipeline-executor.groovy:41]: shouldn't it be `tier-0`? It helps to run against a smoke tested build.

File: pipeline/scheduled-pipeline-executor.groovy [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+def SSHServer = "ssh 10.245.4.89"
+
+// Recipe file path details
+def recipeFile = "RHCEPH-*.yaml"
+def recipeFileDir = "/data/site/recipe"
+
+// Default job parameters
+def buildType = "${params.buildType}" ?: "tier-1"
+def overrides = "${params.overrides}" ?: "{}"
+def tags = "${params.tags}" ?: ""
```
Comment (psathyan) [inline pipeline/scheduled-pipeline-executor.groovy:16]: Shouldn't the default value for tags include `schedule`. This being a scheduled job, I believe we should have it.

File: pipeline/scheduled-pipeline-executor.groovy [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+def tagsList = tags.split(',') as List
+def nodeName = "centos-7"
+if ("ibmc" in tagsList) {
+    nodeName = "agent-01"
+}
```
Comment (psathyan) [inline pipeline/scheduled-pipeline-executor.groovy:21]: I believe this is specific to IBM due to the setting of the default value of `SSHServer`

hence can we rename the script to `ibm-scheduled-pipeline-trigger.groovy`

Comment (vamahaja) [reply pipeline/scheduled-pipeline-executor.groovy:21]: Will add common logic for both IBM and PSI cloud in same script.

File: pipeline/scheduled-pipeline-executor.groovy (hunk 2 above)
Comment (psathyan) [reply pipeline/scheduled-pipeline-executor.groovy:42]: yes as `tier-0` records those that have passed Tier-0 test suites.

File: pipeline/scheduled-pipeline-executor.groovy (hunk 3 above)
Comment (vamahaja) [reply pipeline/scheduled-pipeline-executor.groovy:19]: Done

File: pipeline/scheduled-pipeline-executor.groovy (hunk 2 above)
Comment (vamahaja) [reply pipeline/scheduled-pipeline-executor.groovy]: Done

**Review by psathyan — APPROVED** [review]
Thanks.

**Review by manasagowri — APPROVED** [review]
Looks good to me.

**Review by psathyan — APPROVED** [review]

**Pull request merged**